src/store/index.js

```diff
@@ -85,5 +86,167 @@ export default createStore({
+          localStorage.setItem('allProducts', allProductsListStr)
+        },
+        setPagination (state, page) {
+          const limitLength = 10
+          const totalPages = Math.ceil(state.OriginProductsList.length / limitLength)
+          state.pagination.total_pages = totalPages
+          state.pagination.current_page = page
+          if (state.pagination.current_page === state.pagination.total_pages) {
+            state.pagination.has_next = false
+          } else {
+            state.pagination.has_next = true
+          }
+          if (state.pagination.current_page === 1) {
+            state.pagination.has_pre = false
+          } else {
+            state.pagination.has_pre = true
+          }
+        },
+        perPageProducts (state, perPageProducts) {
+          state.currentPageProducts = perPageProducts
+        },
+        setCurrentPage (state, page) {
+          state.pagination.current_page = page
+        },
+        updateCurrentPageList (state, editedProduct) {
+          state.currentPageProducts = state.currentPageProducts.map(product =>
+            product.id === editedProduct.id ? editedProduct : product
+          )
+        },
+        deleteCurrentPageList (state, id) {
+          state.currentPageProducts = state.currentPageProducts.filter((product) => product.id !== id)
+        }
+      },
+      actions: {
+        setNewProductName ({ commit }, productName) {
+          commit('setNewProductName', productName)
+        },
+        setNewProductPrice ({ commit }, productPrice) {
+          commit('setNewProductPrice', productPrice)
+        },
+        addNewProduct ({ commit }) {
+          if (this.state.products.newProduct.name === '' || this.state.products.newProduct.price <= 0) {
+            alert('請填寫正確的商品資料')
+          } else {
+            commit('setProductId')
+            this.state.products.newProductsList.unshift({ ...this.state.products.newProduct })
+            commit('clearProductInputs')
+            alert('完成新增')
+          }
+        },
+        addToAllProductsList (context) {
+          const { state: { products: { newProductsList, OriginProductsList } } } = this
+          const allProductsList = [...newProductsList, ...OriginProductsList]
+          const allProductsListStr = JSON.stringify(allProductsList)
+          localStorage.setItem('allProducts', allProductsListStr)
+          alert('已送出所有菜單囉')
+          context.commit('clearProductList')
+          context.dispatch('getProducts')
+          context.commit('setPagination')
+        },
+        getProducts (context, page = 1) {
+          const originProducts = JSON.parse(localStorage.getItem('allProducts')) || []
+          // 問題點
+          const perPageProducts = originProducts.filter(function (item, index) {
+            const pageLimit = 10
+            const topPage = (page - 1) * pageLimit
+            const finalPage = page * pageLimit
+
+            return topPage <= index && index < finalPage
+          })
```

Comment:
命名可以讓閱讀更加清晰
舉例:
```suggestion
          const perPageProducts = originProducts.filter(function (product, index) {
            const limit = 10
            const start = (page - 1) * limit
            const end = page * limit

            return start <= index && index < end
          })
```

或是可以寫成:
```
  const limit = 10
  const start = (page - 1) * limit
  const end = page * limit
  const perPageProducts = originProducts.slice(start, end)
```